character_map_tester.h

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright:  Copyright Dimakopoulos Theodoros 2019
+ * License:    $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:    Dimakopoulos Theodoros
+ */
+#pragma once
+// The character map (ch_map) is declared in character_map.h
+
+// Exits the probram 
+// if it encoutners a problem with ch_map

```

Comment:
No need for those 2.

Reply:
Are they obvious? Is it usual to call a function "tester" and mean that it can terminate a program? In my intuition, a tester could do anything from changing the character map to signify an error, to using cerr or cout.

Reply:
> Are they obvious? Is it usual to call a function "tester" and mean that it can terminate a program?

Yes and yes. It's usual for the tester function to be based on `assert()` that if one fails, then the test fails. The reason that these kinds of tests are helpful is that they're automated. You can just run them and not have to do anything yourself to actually test the code.
There's a whole methodology of development called "test-driven development" that is based
mainly on such kinds of tests.
They're not the only way to test code though and it's good to be aware of that as a lot of people are quite passionate about them. We should test whatever fits best our testing purposes (generally for a compiler, these kinds of tests are very convenient).
You may want to check [this video of Jonathan Blow](https://www.youtube.com/watch?v=21JlBOxgGwY) for a "not-so-popular" opinion.
Last note: These kinds of unit tests are usually called unit tests.

>  In my intuition, a tester could do anything from changing the character map to signify an error, to using cerr or cout.
No, generally a tester function does not change the state of the data although this is not a hard rule. It may use output, or errors, but these kinds of tests are again constructed in way
that they are automated. That is, you don't need to see and verify the output yourself.Usually for these tests, there is an output that _you_ know the program must output (this is the correct output). Then, with the test, you test the output the program yielded against this correct output.